### PR TITLE
Add otlploghttp transform package

### PR DESCRIPTION
**Pull request by MrAlias**
Part of #5057 

This does not include testing for `Resource` or `Scope` values. See https://github.com/open-telemetry/opentelemetry-go/issues/5192.

### CODE REVIEW

File: exporters/otlp/otlplog/otlploghttp/exporter.go [hunk 1]
```diff
@@ -47,12 +46,11 @@ func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
...
 	}
-	otlp, err := transformResourceLogs(records)
+	otlp := transformResourceLogs(records)
 	if otlp != nil {
-		// Best effort upload of transformable logs.
-		err = errors.Join(err, e.client.Load().UploadLogs(ctx, otlp))
+		return e.client.Load().UploadLogs(ctx, otlp)
 	}
-	return err
+	return nil
```
Comment (pellared) [inline exporters/otlp/otlplog/otlploghttp/exporter.go:53]: Can we just refactor this to:

```go
if otlp == nil {
  return nil
}

return e.client.Load().UploadLogs(ctx, otlp)
```

File: exporters/otlp/otlplog/otlploghttp/internal/transform/log.go
Comment (pellared) [inline exporters/otlp/otlplog/otlploghttp/internal/transform/log.go:1]: Could we use sync.Pools to reduce the amount of heap allocation?
E.g. `ResourceLogs` could return a functions that would return the pooled maps and slices back to their pools.

If you think that it is possible and it is a good idea then this should be tracked as a separate issue.

**Review by pellared — APPROVED** [review]
LGTM. Just one comment that I find worth addressing.

File: exporters/otlp/otlplog/otlploghttp/exporter.go (hunk 1 above)
Comment (MrAlias) [reply exporters/otlp/otlplog/otlploghttp/exporter.go:53]: I'm not sure how this changes anything, but I can apply it.

Comment (pellared) [reply exporters/otlp/otlplog/otlploghttp/exporter.go]: "usual" code flow is not indented

**Review by pellared — APPROVED** [review]

File: exporters/otlp/otlplog/otlploghttp/exporter.go (hunk 1 above)
Comment (MrAlias) [reply exporters/otlp/otlplog/otlploghttp/exporter.go]: I'm not sure the compiler cares :laughing:

Comment (pellared) [reply exporters/otlp/otlplog/otlploghttp/exporter.go]: It does not. Humans care  😁Similarly to Error Indent Flow

**Review by XSAM — APPROVED** [review]
We might need to move the implementation of transform to a shared place. But we can do that as parts of https://github.com/open-telemetry/opentelemetry-go/issues/5056

~~BTW, this PR seems to be assigned to a wrong project [Go Auto Instrumentation: Beta](https://github.com/orgs/open-telemetry/projects/50) (You might need to move it out of the project, I don't have the write permission of this project)~~

**Pull request merged**